public/js/panier.js

```diff
@@ -1,122 +1,143 @@
 
 
 
-function getLocalStorage() {
-    
 
-    for (let i in basket) {
-        
-        const article = document.createElement("tr");
-        article.classList.add("my-5");
-        article.innerHTML = ('<td> '+ basket[i].Nom +' </td><td>'+ basket[i].Lense +'</td><td>'+ basket[i].Quantity +'</td><td>'+ basket[i].Prix +' €</td>');
 
+// Fonction permettant d"afficher le panier
+function displayLocalStorage() {
+    
+    for (let i in basket) { //Pour chaque élément présent dans le localStorage de la clé "basket"
+        const article = document.createElement("tr"); //Je crée une ligne de tableau
+        article.classList.add("my-5"); 
+        article.innerHTML = ("<td> "+ basket[i].Nom +" </td><td>"+ basket[i].Lense +"</td><td>"+ basket[i].Quantity +"</td><td>"+ basket[i].Prix +" €</td>");
 
-        document.getElementById("display").appendChild(article);
 
-        productsId.push(basket[i].Id)
-        localStorage.setItem("Basketid", JSON.stringify(productsId))
+        document.getElementById("display").appendChild(article); //je l"intègre
 
+        productsId.push(basket[i].Id); //Je met dans le tableau productId son id pour la suite
+        localStorage.setItem("Basketid", JSON.stringify(productsId)); //Je l"écris sur le localStorage avec la clé Basketid
         
-    };
-    console.log(productsId)
+    }
+    console.log(productsId);
 }
 
 
+
+
+
+//Fonction permettant de crée un total de la commande
 function total(){
-    let totalBasket = 0;
+    let totalBasket = 0; //j"initialise la variable a 0(nombre)
 
-    for (let i = 0; i < basket.length; i++){
-        totalBasket += basket[i].Prix * basket[i].Quantity;
+    for (let i = 0; i < basket.length; i++){ //Pour chaque élément présent dans le localStorage de la clé "basket"
+        totalBasket += basket[i].Prix * basket[i].Quantity; //J"incrément avec la somme du prix multiplié avec la quantité
         console.log("totalBasket #" + i + ": " + totalBasket);        
     }
 
-    console.log("totalBasket: " + totalBasket);
-    localStorage.setItem("Prixtotal", totalBasket)
-    document.getElementById("total").innerHTML = totalBasket + ' €'
+    localStorage.setItem("Prixtotal", totalBasket); //Je l"écris sur le localStorage avec la clé "Prixtotal"
+    document.getElementById("total").innerHTML = totalBasket + " €"; //je l"intègre
 }
 
 
+
+
+//Fonction permettant de vider le panier
 function clearBasket(){
     
-    if(localStorage !=0){
-        let button = document.createElement("p");
-        button.classList.add("mb-5");
-        button.innerHTML = ('<a class="bg-dark text-white rounded-pill p-2">vider le panier</a>');
+    if(localStorage !==0){ //si le localStorage est rempli 
+        let button = document.createElement("p"); //Je crée un paragraphe
+        button.classList.add("mb-5"); //Avec cette classe
+        button.innerHTML = ('<button class="bg-dark text-white rounded-pill p-2">vider le panier</button>'); //et ce bouton
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/JulienBrazzalotto/JulienBrazzalotto_5_240921/pullRequest?prid=8291925)
```suggestion
        button.innerHTML = ("<button class=\"bg-dark text-white rounded-pill p-2\">vider le panier</button>"); //et ce bouton
```